src/yunohost/app.py

```diff
@@ -922,7 +918,7 @@ def app_checkurl(auth, url, app=None):
             if path == p:
                 raise MoulinetteError(errno.EINVAL,
                                       m18n.n('app_location_already_used'))
-            elif path.startswith(p):
+            elif path.startswith(p) or p.startswith(path):
                 raise MoulinetteError(errno.EPERM,
                                       m18n.n('app_location_install_failed'))
```

Comment:
`path.startswith(p) or p.startswith(path)` works with all case, even with "/" case ( "/"&"/a/" ,  "/a/"&"/", "/a/b/"&"/a/", "/a/"&"/a/b/").
This line could not be reach if p and path are from the same APPID (see Line 915 `if app is not None and a['id'] == app:`) .